### DISCUSSION TITLE
Issues with "brew install gnutls" on Mac Studio libtool segfault 11

### DISCUSSION THREAD

**Discussion by jerrynsr** [Homebrew/discussions#3147] [Writing Formulae/Casks]
Have any of you attempted to install OpenCV with Hombrew on a Mac Studio ?  Libtool segfaults when building sha1-armv8.o

**Comment by SMillerDev** [comment]
What do `brew doctor` and `brew config` say?

**Reply by jerrynsr** [reply]
I have Brew installed in /opt/homebrew
Here's my symbolic link:
jerryn@Mac-Studio / % ls -l /opt
total 0
drwxr-xr-x  9 root    wheel  288 Apr 25  2021 X11
lrwxr-xr-x  1 jerryn  admin   26 Apr  3 16:06 homebrew -> /Volumes/Macports/homebrew

jerryn@Mac-Studio / % brew doctor   
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Homebrew's prefix is not /opt/homebrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/opt/homebrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.

Here's brew config:
jerryn@Mac-Studio / % brew config
HOMEBREW_VERSION: 3.4.5-16-gf6ab300
ORIGIN: https://github.com/Homebrew/brew
HEAD: f6ab300fc169777705f025c67c4b7c35605f4a66
Last commit: 27 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5d0304540ae81778bd58308375e4432bba0fa021
Core tap last commit: 51 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /Volumes/Macports/homebrew
HOMEBREW_REPOSITORY: /Volumes/Macports/homebrew
HOMEBREW_CELLAR: /Volumes/Macports/homebrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.sulUuTrnmf/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_TEMP: /opt/homebrew/temp
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-arm64
CLT: 13.3.0.0.1.1645755326
Xcode: 13.3
Rosetta 2: false

**Reply by jerrynsr** [reply]
> What do `brew doctor` and `brew config` say?

For now I was able to install with --force-bottle.    I am installing opencv now.  Hopefully it will continue.  Maybe something broke with the MacOS 12.3.1 SDK ?

**Reply by SMillerDev** [reply]
It's more likely related to the non-standard Homebrew prefix. Just creating a symlink doesn't really resolve those issues.

**Comment by jay7x** [comment]
I just got the same issue while upgrading gnutls (3.7.3 -> 3.7.4) on M1 Pro Macbook:
```
../../../libtool: line 1760: 60962 Segmentation fault: 11  clang -Wa,-march=all -Wno-implicit-function-declaration -c macosx/sha1-armv8.s -fno-common -DPIC -o macosx/.libs/sha1-armv8.o
../../../libtool: line 1760: 60966 Segmentation fault: 11  clang -Wa,-march=all -Wno-implicit-function-declaration -c macosx/sha256-armv8.s -fno-common -DPIC -o macosx/.libs/sha256-armv8.o
make[4]: *** [macosx/sha1-armv8.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
../../../libtool: line 1760: 60987 Segmentation fault: 11  clang -Wa,-march=all -Wno-implicit-function-declaration -c macosx/sha512-armv8.s -fno-common -DPIC -o macosx/.libs/sha512-armv8.o
make[4]: *** [macosx/sha512-armv8.lo] Error 1
make[4]: *** [macosx/sha256-armv8.lo] Error 1
../../../libtool: line 1760: 60973 Segmentation fault: 11  clang -Wa,-march=all -Wno-implicit-function-declaration -c macosx/aes-aarch64.s -fno-common -DPIC -o macosx/.libs/aes-aarch64.o
make[4]: *** [macosx/aes-aarch64.lo] Error 1
make[3]: *** [install-recursive] Error 1
make[2]: *** [install-recursive] Error 1
make[1]: *** [install] Error 2
make: *** [install-recursive] Error 1
```

P.S. My homebrew dir is located under `~/.homebrew` folder.

**Reply by fionahiklas** [reply]
I'm setting the same result with the homebrew directory in various locations.  If I instead follow the convention and use `/opt/homebrew` the installation of `gnutls` works fine.  I suspect that, in this case, it's just uncompressing an existing build into that location rather than trying to compile from scratch, it's alot faster.

It looks like the libtool that is breaking is the one built by `brew` but I'm not familiar with how brew builds packages or were the code actually is.  If someone can point me in the right direction it would probably be worth just looking at line 1760 to so what is going wrong in libtool.  While using `/opt/homebrew` works I wonder if that is just avoiding the problem rather than fixing it?

**Reply by fionahiklas** [reply]
Actually, I'm digging a little more and I'm less sure it's the brew libtool but might be the MacOS native one.

**Comment by Bo98** [comment]
This is a Xcode 13.3 compiler crash bug. I will try workaround it however: https://github.com/Homebrew/homebrew-core/pull/99002

**Reply by fionahiklas** [reply]
The above PR is still waiting to be merged (there appears to be some issue with tests failing that don't like related to the actual change) but I manually applied the edits https://github.com/Homebrew/homebrew-core/pull/99002/files to the `gnutls.rb` file and re-ran `brew install gnutls` 

My homebrew install is under `$HOME/tools/homebrew` so the location of the formula is `$HOME/tools/homebrew/Library/Taps/homebrew/homebrew-core/Formula/gnutls.rb` and I just backed up that file and then carried out the edits as per the diffs - I probably could have just downloaded the copy of the file from the PR instead :)
